### ISSUE TITLE
Impossible to jump obstacles, some weird behaviour while loading and problems with rectangles collisions

### ISSUE THREAD

**Issue by bydariogamer** [pserwylo/beat-game#53]
There are impossible to jump obstacles. It is not a big big problem, because as all the higher obstacles are streetlights, there is not a lot of damage. However, it is wrong. It is in the Laundry Room level.
The first time I played the level it started loading the Laundry Room and then the Ballroom. When the game started it was the Laundry level (or maybe 'song' is more correct ;) ), some seconds later it changed to the Ballroom, and some seconds later the Laundry... After it continues as it should.
Finally, there is a problem with the corners of the platform buildings and the jump. I think you may have a _bullet through paper_ bug... When the player is descending from a high jump and it touches the end corner of a building platform, the game puts it back in the corner, but it is damaged... Maybe the order in which the updates are made to the player.

**Comment by pserwylo** [comment]
Thanks again for the feedback:

> There are impossible to jump obstacles. It is not a big big problem, because as all the higher obstacles are streetlights, there is not a lot of damage. However, it is wrong.

Yes, this is something I was aware of. In v0.2.0 it is less of an issue though, so still keep an eye out for that ;) Having said that, I didn't implement anything to specifically deal with it, just that the new algorithm seems to be much less susceptible to it.

> It is in the Laundry Room level. The first time I played the level it started loading the Laundry Room and then the Ballroom. When the game started it was the Laundry level (or maybe 'song' is more correct ;) ), some seconds later it changed to the Ballroom, and some seconds later the Laundry... After it continues as it should.

Another v0.1.0 bug, which should be addressed in the new v0.2.0. This was fixed when changing the menu to deal with all 17 levels in the new version :)

> Finally, there is a problem with the corners of the platform buildings and the jump. I think you may have a bullet through paper bug... When the player is descending from a high jump and it touches the end corner of a building platform, the game puts it back in the corner, but it is damaged... Maybe the order in which the updates are made to the player.

Hmm, so this is either a bug or part of the intentional behaviour. There is an expectation that "near enough is good enough" here, because sub pixel collisions in such a frantic game would result in a pretty un-fun game. Therefore, there is a notion of `CLIMB_THRESHOLD` in the code: https://github.com/pserwylo/beat-game/blob/master/core/src/com/serwylo/beatgame/entities/Player.kt#L266.

Also, I tweaked the algorithm a little in v0.2.0 (or perhaps v0.3.0 - can't recall) to treat the character as 4/5ths the width they actually are, because their feet are not to the far left or right of the character sprite. This resulted in much nicer looking physics when walking off the edge of a building.

However, I'm also aware that there is some bugs with relation to dropping through the edge of the world if you drop down too fast. If that is the case, then lets open a new issue to address that.
